Status page

DamianEdwards opened the pull request:
Added a status page that can be looked at by humans or remote monitoring services to see the state of the various JabbR sub systems, e.g. database, SignalR, Azure blog storage, etc.

DamianEdwards committed:
Added status page at ~/status
- Checks if Hub message can be sent
- Checks if DB can be queried

DamianEdwards committed:
More status page updates
- Moved to home module
- Pass details of systems status to view and render in a table

DamianEdwards committed:
Status page changes:
- Check status of Azure blob storage if configured
- Set view status code to 500 if there are any errors
- Support forcing an error via the querystring: ~/status?fail=1 (good for testing)
- Refactored the model to have strongly typed SystemStatus entries instead of dictionary
- #1015

davidfowl commented on JabbR/Nancy/HomeModule.cs:
```diff
@@ -63,6 +64,89 @@ public class HomeModule : JabbRModule
+                }
+                catch (Exception ex)
+                {
+                    signalrStatus.SetException(ex);
```
It's going to be aggregate.

davidfowl commented on JabbR/Nancy/HomeModule.cs:
```diff
@@ -63,6 +64,89 @@ public class HomeModule : JabbRModule
+                }
+                catch (Exception ex)
+                {
+                    azureStorageStatus.SetException(ex);
```
Same here. Unwrap

davidfowl commented on JabbR/ViewModels/StatusViewModel.cs:
```diff
@@ -0,0 +1,26 @@
+        {
+            get
+            {
+                return !_systems.Any(s => !(s.IsOK ?? true));
```
What the what. Please change it.

``` csharp
_system.All(s => s.IsOk);
```

DamianEdwards committed:
Address review feedback:
- Unwrap exception
- Use simpler but slower method to get overall status bool
- #1016

davidfowl commented:
:shipit:
